src/main/cpp/win_fsnotifier.cpp

```diff
@@ -0,0 +1,209 @@
+// Initial version copied from Initial version copied from https://github.com/JetBrains/intellij-community/blob/master/native/WinFsNotifier/fileWatcher3.c
+// Copyright 2000-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
+
+#ifdef _WIN32
+
+#include "native.h"
+#include "generic.h"
+#include "win.h"
+
+JavaVM* jvm = NULL;
+
+typedef struct watch_details {
+    HANDLE threadHandle;
+    HANDLE stopEventHandle;
+    wchar_t drivePath[4];
+    int watchedPathCount;
+    wchar_t **watchedPaths;
+    jobject watcherCallback;
+} watch_details_t;
+
+// TODO Find the right size for this
+#define EVENT_BUFFER_SIZE (16*1024)
+
+#define CREATE_SHARE (FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE)
+#define CREATE_FLAGS (FILE_FLAG_BACKUP_SEMANTICS | FILE_FLAG_OVERLAPPED)
+#define EVENT_MASK (FILE_NOTIFY_CHANGE_FILE_NAME | FILE_NOTIFY_CHANGE_DIR_NAME | \
+                    FILE_NOTIFY_CHANGE_ATTRIBUTES | FILE_NOTIFY_CHANGE_SIZE | FILE_NOTIFY_CHANGE_LAST_WRITE)
+
+void handlePathChanged(watch_details_t *details, FILE_NOTIFY_INFORMATION *info) {
+    // const char *event;
```

Comment:
Can you add a TODO/remove the commented out code?

Reply:
I think we are at a different level of completeness here than to worry about removing commented-out code.